src/compiler/transformers/declarations/emitResolver.ts

```diff
@@ -0,0 +1,675 @@
+import {
+    AnyImportSyntax,
+    appendIfUnique,
+    ComputedPropertyName,
+    createEvaluator,
+    Debug,
+    Declaration,
+    DeclarationName,
+    determineIfDeclarationIsVisible,
+    ElementAccessExpression,
+    emptyArray,
+    EntityNameOrEntityNameExpression,
+    EnumDeclaration,
+    EnumMember,
+    every,
+    Expression,
+    factory,
+    filter,
+    findAncestor,
+    FunctionDeclaration,
+    FunctionLikeDeclaration,
+    getFirstIdentifier,
+    getNameOfDeclaration,
+    getParseTreeNode,
+    getTextOfNode,
+    hasDynamicName,
+    hasProperty,
+    hasSyntacticModifier,
+    Identifier,
+    ImportClause,
+    ImportEqualsDeclaration,
+    ImportSpecifier,
+    isAccessor,
+    isArrowFunction,
+    isBigIntLiteral,
+    isBinaryExpression,
+    isBindingElement,
+    isDeclarationReadonly,
+    isElementAccessExpression,
+    isEntityNameExpression,
+    isEnumDeclaration,
+    isEnumMember,
+    isExpressionStatement,
+    isFunctionDeclaration,
+    isFunctionExpression,
+    isFunctionLike,
+    isGetAccessor,
+    isGetAccessorDeclaration,
+    isIdentifier,
+    isInfinityOrNaNString,
+    isInJSFile,
+    isLateVisibilityPaintedStatement,
+    isNumericLiteral,
+    isPartOfTypeNode,
+    isPrefixUnaryExpression,
+    isPropertyAccessExpression,
+    isSetAccessor,
+    isSetAccessorDeclaration,
+    isStringLiteralLike,
+    isTemplateExpression,
+    isThisIdentifier,
+    isVarConst,
+    isVariableDeclaration,
+    isVariableStatement,
+    LateBoundDeclaration,
+    LateVisibilityPaintedStatement,
+    ModifierFlags,
+    NamespaceImport,
+    Node,
+    NodeArray,
+    NodeFlags,
+    nodeIsPresent,
+    NoSubstitutionTemplateLiteral,
+    ParameterDeclaration,
+    parsePseudoBigInt,
+    PropertyAccessExpression,
+    PropertyDeclaration,
+    PropertyName,
+    PropertySignature,
+    skipParentheses,
+    SourceFile,
+    Statement,
+    SymbolAccessibility,
+    SymbolFlags,
+    SymbolVisibilityResult,
+    SyntaxKind,
+    VariableDeclaration,
+} from "../../_namespaces/ts";
+import {
+    bindSourceFileForDeclarationEmit,
+    EmitDeclarationNodeLinks,
+    EmitDeclarationSymbol,
+} from "./emitBinder";
+import {
+    IsolatedEmitResolver,
+    MemberKey,
+} from "./types";
+import {
+    getMemberKey,
+} from "./utils";
+
+const knownFunctionMembers = new Set([

```

Comment:
This is the explanation, I will add it as a comment: 

Assigning values to a property of a function will usually cause those members to be implicitly declared on the function
even if they were were not declared (expando functions)
DTE needs to detect these members and error on them since this behavior is not supported in isolated declarations
There are however members that can be assigned on a function that are not expando members, namely members that come from Function
In DTE we do not load the full lib.d.ts so we keep a list of known members of function that can be assigned without considering them expando members.